### ISSUE TITLE
unarchive: unhelpful error message when "owner" does not exist and unarchiving tar.gz

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29455]
From @bart2 on 2016-11-29T15:31:14Z
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
unarchive

##### ANSIBLE VERSION

```
ansible 2.3.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### OS / ENVIRONMENT

N/A

##### SUMMARY

When "owner" parameter is set to user that does not exist, and we are extracting tar.gz archive the module fails with unrelated error message: "Unexpected error when accessing exploded file: [Errno 2] No such file or directory:"

Interesting note: when the file being archived has no extension, ie "test" instead of "test.txt" below, the issue does not occur.

##### STEPS TO REPRODUCE

```
mkdir /tmp/unarchive_bug

cd /tmp/unarchive_bug

mkdir unarchived

echo "test" > test.txt
tar cf testfile.tar test.txt
gzip -9 testfile.tar

cat <<'EOF' > unarchive_test.yml
- hosts: localhost
  tasks:
    - name: unarchive test
      unarchive:
        copy: no
        src: /tmp/unarchive_bug/testfile.tar.gz
        dest: /tmp/unarchive_bug/unarchived
        owner: doesnotexist
EOF

ansible-playbook -c local unarchive_test.yml
```

##### EXPECTED RESULTS

Failure message that indicates that user "doesnotexist" does not exist.

##### ACTUAL RESULTS

```
TASK [unarchive test] **********************************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "dest": "/tmp/unarchive_bug/unarchived", "failed": true, "gid": 0, "group": "root", "handler": "TgzArchive", "mode": "0755", "msg": "Unexpected error when accessing exploded file: [Errno 2] No such file or directory: '/tmp/unarchive_bug/unarchived/test.txt'", "owner": "root", "size": 21, "src": "/tmp/unarchive_bug/testfile.tar.gz", "state": "directory", "uid": 0}
	to retry, use: --limit @/tmp/unarchive_bug/unarchive_test.retry
```

Copied from original issue: ansible/ansible-modules-core#5760

**Comment by ansibot** [comment]
From @ansibot on 2016-11-29T15:31:14Z


@dagwieers, @pileofrogs, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by toadzky** [comment]
this bug applies to non-existent groups too

**Comment by Akasurde** [comment]
@bart2 @toadzky I am unable to reproduce this issue against 2.4 version. 

```
# ansible --version
ansible 2.4.3.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.13 (default, May 10 2017, 20:04:28) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]

# ansible-playbook -c local unarchive_test.yml
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'


PLAY [localhost] ***************************************************************************************************************

TASK [Gathering Facts] *********************************************************************************************************
ok: [localhost]

TASK [unarchive test] **********************************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "gid": 0, "group": "root", "mode": "0644", "msg": "chown failed: failed to look up user doesnotexist", "owner": "root", "path": "/tmp/unarchive_bug/unarchived/test.txt", "size": 5, "state": "file", "uid": 0}
	to retry, use: --limit @/tmp/unarchive_bug/unarchive_test.retry

PLAY RECAP *********************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1
```

Could you please check if this is the case for you ?
needs_info

**Comment by ansibot** [comment]
@bart2 This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@bart2 You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
